pkg/theia/commands/policy_recommendation_status.go

```diff
@@ -15,243 +15,81 @@
-	sparkApplication, err := getSparkAppByRecommendationID(clientset, id)
+	err = ParseRecommendationName(prName)
 	if err != nil {
-		return "", err
+		return err
 	}
-	errorMessage := strings.TrimSpace(string(sparkApplication.Status.AppState.ErrorMessage))
-	return errorMessage, nil
-}
-
-func getPolicyRecommendationProgress(baseUrl string) (string, error) {
-	// Get the id of current Spark application
-	url := fmt.Sprintf("%s/api/v1/applications", baseUrl)
-	response, err := getResponseFromSparkMonitoringSvc(url)
+	useClusterIP, err := cmd.Flags().GetBool("use-cluster-ip")
 	if err != nil {
-		return "", fmt.Errorf("failed to get response from the Spark Monitoring Service: %v", err)
+		return err
 	}
-	var getAppsResult []map[string]interface{}
-	json.Unmarshal([]byte(response), &getAppsResult)
-	if len(getAppsResult) != 1 {
-		return "", fmt.Errorf("wrong Spark Application number, expected 1, got %d", len(getAppsResult))
-	}
-	sparkAppID := getAppsResult[0]["id"]
-	// Check the percentage of completed stages
-	url = fmt.Sprintf("%s/api/v1/applications/%s/stages", baseUrl, sparkAppID)
-	response, err = getResponseFromSparkMonitoringSvc(url)
+	theiaClient, pf, err := SetupTheiaClientAndConnection(cmd, useClusterIP)
 	if err != nil {
-		return "", fmt.Errorf("failed to get response from the Spark Monitoring Service: %v", err)
-	}
-	var getStagesResult []map[string]interface{}
-	json.Unmarshal([]byte(response), &getStagesResult)
-	NumStageResult := len(getStagesResult)
-	if NumStageResult < 1 {
-		return "", fmt.Errorf("wrong Spark Application stages number, expected at least 1, got %d", NumStageResult)
+		return fmt.Errorf("couldn't setup theia client: %v", err)
 	}
-	completedStages := 0
-	for _, stage := range getStagesResult {
-		if stage["status"] == "COMPLETE" || stage["status"] == "SKIPPED" {
-			completedStages++
-		}
+	if pf != nil {
+		defer pf.Stop()
 	}
-	return fmt.Sprintf(": %d/%d (%d%%) stages completed", completedStages, NumStageResult, completedStages*100/NumStageResult), nil
-}
-
-func getResponseFromSparkMonitoringSvc(url string) ([]byte, error) {
-	sparkMonitoringClient := http.Client{}
-	request, err := http.NewRequest(http.MethodGet, url, nil)
+	npr, err := getPolicyRecommendationByName(theiaClient, prName)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("error when getting policy recommendation job by using job name: %v", err)
 	}
-	var res *http.Response
-	var getErr error
-	connRetryInterval := 1 * time.Second
-	connTimeout := 10 * time.Second
-	if err := wait.PollImmediate(connRetryInterval, connTimeout, func() (bool, error) {
-		res, err = sparkMonitoringClient.Do(request)
-		if err != nil {
-			getErr = err
-			return false, nil
+	state := npr.Status.State
+	if state == "RUNNING" {
+		completedStages := npr.Status.CompletedStages
+		totalStages := npr.Status.TotalStages
+		if totalStages < 1 {
+			return fmt.Errorf("wrong Spark Application stages number, expected at least 1, got %d", totalStages)
```

Comment:
I'm thinking about not to display this error to users. At the beginning of the Spark Application, the totalStages will be 0 for a while. In this case, we can display a progress of `0/0 0%` or just display the RUNNING state without the progress.

And even if we plan to display this error, I think we still need to display the state.

Reply:
I fixed it to display the progress as 0/0 0% when total stage is zero. Thanks.